homeassistant/components/homekit_controller/air_quality.py

```diff
@@ -20,6 +24,21 @@
 class HomeAirQualitySensor(HomeKitEntity, AirQualityEntity):
     """Representation of a HomeKit Controller Air Quality sensor."""
 
+    def __init__(self, *args, **kwargs):
+        """Initialise a HomeKit air_quality acessory."""
+        super().__init__(*args, **kwargs)
+
+        _LOGGER.warning(
+            "The homekit_controller air_quality entity has been "
+            "deprecated and will be removed in 2021.12.0"
+        )
```

Comment:
Can you avoid the side effect in `__init__`?

Reply:
I mean there are other places I could put the logging I guess. If i put it in `added_to_hass` would that work? I guess it wouldn't trigger the warning if you turned off the entity in the entity registry in that case.

Reply:
Warning moved to `async_added_to_hass`.